.github/workflows/ci.yaml

```diff
@@ -21,21 +21,35 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - scala_version: "2.12.15"
-            spark_version: "3.3.4"
-          - scala_version: "2.12.18"
-            spark_version: "3.5.5"
+         -  spark_version: "3.3.4"
+            scala_version: "2.12.15"
+         -  spark_version: "3.4.4"
+            scala_version: "2.12.17"
+         -  spark_version: "3.5.5"
+            scala_version: "2.12.18"
 
-          - scala_version: "2.13.8"
-            spark_version: "3.3.4"
-          - scala_version: "2.13.8"
-            spark_version: "3.5.5"
-          - scala_version: "2.13.8"
-            spark_version: "4.1.0-SNAPSHOT"
+         -  spark_version: "3.3.4"
+            scala_version: "2.13.8"
+         -  spark_version: "3.4.4"
+            scala_version: "2.13.8"
+         -  spark_version: "3.5.5"
+            scala_version: "2.13.8"
+         -  spark_version: "4.0.1-SNAPSHOT"
+            scala_version: "2.13.8"
```

Comment:
Right now, we don't have good integration tests that confirm that all these different version work, (even if they happen to build).   Those tests will hopefully get added as a part of: https://github.com/G-Research/spark/issues/45

I have run the demo on 3.3.4, 3.5.3 and 4.1.0 snapshot with the default scala versions, so I believe those to work.

I'm hesitant to add all these other versions until they are tested, because it will be harder for people to know what works.  What do you think?

Reply:
For the time being, we can setup the demo to run as integration tests in the CI.

Reply:
I have moved the new Spark versions into #11.